runtime/src/accounts_db.rs

```diff
@@ -1464,39 +1464,52 @@ impl AccountsDb {
         }
     }
 
-    fn collect_uncleaned_pubkeys_to_slot(&self, max_slot: Slot) -> (Vec<Vec<Pubkey>>, Slot) {
-        let mut max_slot_in_uncleaned_pubkeys = 0;
-        let slots: Vec<Slot> = self
-            .uncleaned_pubkeys
+    /// Collect all the uncleaned slots, up to a max slot
+    ///
+    /// Search through the uncleaned Pubkeys and return all the slots, up to a maximum slot.
+    fn collect_uncleaned_slots_up_to_slot(&self, max_slot: Slot) -> Vec<Slot> {
+        self.uncleaned_pubkeys
             .iter()
             .filter_map(|entry| {
-                let slot = entry.key();
-                max_slot_in_uncleaned_pubkeys = max_slot_in_uncleaned_pubkeys.max(*slot);
-                if *slot <= max_slot {
-                    Some(*slot)
+                let slot = *entry.key();
+                (slot <= max_slot).then(|| slot)
```

Comment:
maybe: .then(|| *entry.key()) and delete line 1474

Reply:
I think I could do
```
(entry.key() <= max_slot).then(|| *entry.key())
```
to get it onto a single line. I thought it was a bit easier to read with the `let slot = ...` so that (1) I didn't have to duplicate `entry.key()`, and (2) so that it is clear what `entry.key()` is. Thoughts?

Reply:
yep, I agree. I didn't realize it was used twice.